components/dashboard/src/components/IdeOptions.tsx

```diff
@@ -0,0 +1,283 @@
+/**
+ * Copyright (c) 2024 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { useEffect, useMemo, useState } from "react";
+import { LoadingButton } from "@podkit/buttons/LoadingButton";
+import { Button } from "@podkit/buttons/Button";
+import { SwitchInputField } from "@podkit/switch/Switch";
+import { cn } from "@podkit/lib/cn";
+import { UseMutationResult } from "@tanstack/react-query";
+import { MiddleDot } from "./typography/MiddleDot";
+import { useToast } from "./toasts/Toasts";
+import Modal, { ModalBaseFooter, ModalBody, ModalHeader } from "./Modal";
+import { LoadingState } from "@podkit/loading/LoadingState";
+import { IDEOption, IDEOptions } from "@gitpod/gitpod-protocol/lib/ide-protocol";
+import { getGitpodService } from "../service/service";
+
+interface IdeOptionsProps {
+    ideOptions: IDEOptions | undefined;
```

Comment:
to be consistent with how we do it just below

```suggestion
    ideOptions?: IDEOptions;
```

Reply:
Not the same, using `?` makes it as optional as a prop, using `| undefined` it's required to be set even if it's `undefined`

Reply:
I understand the technical difference, but why would we want to differentiate between the two? If it's fine to provide `undefined`, why do we force it to be provided?